src/components/Spinner.js

```diff
@@ -0,0 +1,49 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import { connectTheme } from '../utils';
+
+class Spinner extends PureComponent {
+  static get propTypes() {
+    return {
+      className: PropTypes.string,
+      style: PropTypes.object,
+      theme: PropTypes.object,
+      error: PropTypes.object,
+      size: PropTypes.string,
+    };
+  }
+
+  render() {
+    const {
+      className = '',
+      style: _style,
+      // fontawesome size suffixes https://fontawesome.com/how-to-use/on-the-web/styling/sizing-icons
+      size = '2x',
+      theme: { themeVariables: { themeColors } },
+      error,
+      ...otherProps
+    } = this.props;
+
+    const style = { color: themeColors.primary, ..._style };
+    let loader;
+    if (error) {
+      loader = (
+        <div className="error" style={style} {...otherProps}>
```

Comment:
`div` vs `span`?

Reply:
hmmm, good question. I think I like `div` for this because a spinner you kind of want to be in its own box, but I suppose the argument could be made that should be left up to the implementer? I think I like this better though so that it will work out of the box and then can be changed via style prop but I'm open to the argument otherwise.

Reply:
Maybe its premature optimization